lib/AST/CASTBridging.cpp

```diff
@@ -255,104 +314,113 @@ void *FunctionCallExpr_create(BridgedASTContext cContext, void *fn,
-                            BridgedSourceLoc cLoc) {
-  ASTContext &context = convertASTContext(cContext);
-  auto name = DeclNameRef{convertIdentifier(base)};
-  Expr *E = new (context) UnresolvedDeclRefExpr(
-      name, DeclRefKind::Ordinary, DeclNameLoc{convertSourceLoc(cLoc)});
-  return E;
+BridgedUnresolvedDeclRefExpr UnresolvedDeclRefExpr_createParsed(
+    BridgedASTContext cContext, BridgedIdentifier base, BridgedSourceLoc cLoc) {
+  ASTContext &context = unbridged(cContext);
+  auto name = DeclNameRef{unbridged(base)};
+  auto *E = new (context) UnresolvedDeclRefExpr(name, DeclRefKind::Ordinary,
+                                                DeclNameLoc{unbridged(cLoc)});
+  return bridged(E);
 }
 
-void *StringLiteralExpr_create(BridgedASTContext cContext, BridgedString cStr,
+BridgedStringLiteralExpr
+StringLiteralExpr_createParsed(BridgedASTContext cContext, BridgedString cStr,
                                BridgedSourceLoc cTokenLoc) {
-  ASTContext &context = convertASTContext(cContext);
-  auto str = context.AllocateCopy(convertString(cStr));
-  return new (context) StringLiteralExpr(str, convertSourceLoc(cTokenLoc));
+  ASTContext &context = unbridged(cContext);
+  auto str = context.AllocateCopy(unbridged(cStr));
+  return bridged(new (context) StringLiteralExpr(str, unbridged(cTokenLoc)));
 }
 
-void *IntegerLiteralExpr_create(BridgedASTContext cContext, BridgedString cStr,
+BridgedIntegerLiteralExpr
+IntegerLiteralExpr_createParsed(BridgedASTContext cContext, BridgedString cStr,
                                 BridgedSourceLoc cTokenLoc) {
-  ASTContext &context = convertASTContext(cContext);
-  auto str = context.AllocateCopy(convertString(cStr));
-  return new (context) IntegerLiteralExpr(str, convertSourceLoc(cTokenLoc));
+  ASTContext &context = unbridged(cContext);
+  auto str = context.AllocateCopy(unbridged(cStr));
+  return bridged(new (context) IntegerLiteralExpr(str, unbridged(cTokenLoc)));
 }
 
-void *ArrayExpr_create(BridgedASTContext cContext, BridgedSourceLoc cLLoc,
-                       BridgedArrayRef elements, BridgedArrayRef commas,
-                       BridgedSourceLoc cRLoc) {
-  ASTContext &context = convertASTContext(cContext);
-  return ArrayExpr::create(
-      context, convertSourceLoc(cLLoc), convertArrayRef<Expr *>(elements),
-      convertArrayRef<SourceLoc>(commas), convertSourceLoc(cRLoc));
+BridgedArrayExpr ArrayExpr_createParsed(BridgedASTContext cContext,
+                                        BridgedSourceLoc cLLoc,
+                                        BridgedArrayRef elements,
+                                        BridgedArrayRef commas,
+                                        BridgedSourceLoc cRLoc) {
+  ASTContext &context = unbridged(cContext);
+  return bridged(ArrayExpr::create(
+      context, unbridged(cLLoc), unbridgedArrayRef<Expr *>(elements),
+      unbridgedArrayRef<SourceLoc>(commas), unbridged(cRLoc)));
```

Comment:
Not really important but do we have a somewhat consistent style if we do `return bridged(XXX::create)` or 
```cpp
auto X = XXX::create…`
return bridged(X);
```

I would have probably used the latter version because it means that the parameters aren’t nested as much. But it’s not all that important.

Reply:
I think for single-line cases, nesting it in the `bridged(...)` call is fine, but yeah maybe we should split out the multi-line cases